content/src/content/jcr_root/apps/core/wcm/components/carousel/v1/carousel/clientlibs/site/js/carousel.js

```diff
@@ -145,6 +153,29 @@
             }
         }
 
+        /**
+         * Displays the slide containing the element that corresponds to the deep link in the URI fragment
+         * and scrolls the browser to this element.
+         */
+        function scrollToDeepLinkIdInCarousel() {
+            if (containerUtils) {
+                var deepLinkItemIdx = containerUtils.getDeepLinkItemIdx(that, "item", "item");
+                if (deepLinkItemIdx && deepLinkItemIdx !== -1) {
+                    var deepLinkItem = that._elements["item"][deepLinkItemIdx];
+                    if (deepLinkItem && that._elements["item"][that._active].id !== deepLinkItem.id) {
+                        navigateAndFocusIndicator(deepLinkItemIdx, true);
+                        // pause the carousel auto-rotation
+                        pause();
+                    }
+                    var hashId = window.location.hash.substring(1);
+                    var hashItem = document.querySelector("[id='" + hashId + "']");
```

Comment:
See comment above regarding empty hashId

Reply:
Addressed in https://github.com/adobe/aem-core-wcm-components/pull/2138/commits/5b6735af3ae0bf75e885c69d9bf51dc94cd643be